volttron/utils/rmq_setup.py

```diff
@@ -142,51 +194,61 @@ def _set_initial_rabbit_config(instance_name):
                                                        config=config_opts)
         config_opts.sync()
     else:
-        config_opts['user'] = "guest"
-        prompt = 'What is the password for RabbitMQ default guest user?'
-        new_pass = prompt_response(prompt, default="guest")
-        config_opts['pass'] = "guest"
-
-        prompt = 'What is the hostname of system?'
-        new_host = prompt_response(prompt, default='locahost')
-        config_opts['host'] = new_host
-        # TODO - How to configure port other than 5671 for ssl - validate should
-        # check if port is not 5672.
-        port = config_opts.get('amqp-port', 5672)
         prompt = 'What is the instance port for the RabbitMQ address?'
-        valid_port = False
-        while not valid_port:
-            port = prompt_response(prompt, default=port)
-            valid_port = is_valid_amqp_port(port)
-            if not valid_port:
-                print("Port is not valid")
-        config_opts['amqp-port'] = str(port)
-
-        port = config_opts.get('mgmt-port', 15672)
+        prompt_port(config_opts, 'amqp-port', 5672, prompt)
+
         prompt = 'What is the instance port for the RabbitMQ management plugin?'
+        prompt_port(config_opts, 'mgmt-port', 15672, prompt)
         valid_port = False
-        while not valid_port:
-            port = prompt_response(prompt, default=port)
-            valid_port = is_valid_mgmt_port(port)
-            if not valid_port:
-                print("Port is not valid")
-        config_opts['mgmt-port'] = str(port)
 
         config_opts.sync()
-        #print config_opts
+        new_conf = """listeners.tcp.default = {}
+        management.listener.port = {}""".format(config_opts['amqp-port'],
+                                                config_opts['mgmt-port'])
+        with open(os.path.join(config_opts['rmq-home'],
+                               "etc/rabbitmq", "rabbitmq.conf"),
+                  'w+') as r_conf:
+            r_conf.write(new_conf)
+
+
+    # Start rabbitmq server
+    start_rabbit(config_opts['rmq-home'])
+
+    #enable plugins
+    cmd =[os.path.join(config_opts['rmq-home'], "sbin/rabbitmq-plugins"),
+                       "enable", "rabbitmq_management",
+                       "rabbitmq_federation",
+                       "rabbitmq_federation_management",
+                       "rabbitmq_shovel", "rabbitmq_auth_mechanism_ssl"]
+    subprocess.check_call(cmd)
+    print("**enabled plugins")
+
+
+def prompt_port(config_opts, config_key, default_port, prompt):
+    # TODO - How to configure port other than 5671 for ssl - validate should
+    # check if port is not 5672.
+    port = config_opts.get(config_key, default_port)
+    valid_port = False
+    while not valid_port:
+        port = prompt_response(prompt, default=port)
+        valid_port = is_valid_amqp_port(port)
+        if not valid_port:
+            print("Port is not valid")
+    config_opts['amqp-port'] = str(port)
```

Comment:
It should be updated based on config key

Reply:
Fixed

Reply:
fixed